eng/vmr-sync.sh

```diff
@@ -201,9 +213,11 @@ else
     exit 1
   fi
 
-  highlight "Preparing $vmr_dir"
-  git -C "$vmr_dir" checkout "$vmr_branch"
-  git -C "$vmr_dir" pull
+  if [[ "$no_vmr_prepare" == "false" ]]; then
```

Comment:
Shouldn't you use single '=' for string comparison? The double equal sign might be possible for double "[[" bracket Korn-shell extension tests (as opposed to single one), but it's worth double-checking.

Reply:
Seems like it's `==` for both `[` and `[[`. I can't remember these things whether with `[` you need `-eq` or `=` or `==` so I always use `[[` and remember that one is modern like other languages. I understand this can technically not work in some Linux distros but not sure the probability is high?

Reply:
The difference in bash is nicely described for example here: https://askubuntu.com/questions/445749/whats-the-difference-between-shell-builtin-and-shell-keyword. However, this is different in plain bourne shell (sh) or other shells, where '[[' is not defined at all. This might bite you if you sometimes run commands through e.g., 'sh -c'. That's why it's usually better to use subshell (parentheses) for the call as you are guaranteed the subshell process will be run with the same shell as the calling process.